### PR TITLE
[WFCORE-5181] Expression properties with trailing whitespaces are resolved to a trimmed value

### PR DESCRIPTION
**Pull request by jbaesner**
Issue: https://issues.redhat.com/browse/WFCORE-5181

### CODE REVIEW

**Comment by spyrkob** [comment]
@jmesnil could you review this when you have a moment please?

File: controller/src/main/java/org/jboss/as/controller/parsing/ParseUtils.java [hunk 1]
```diff
@@ -565,8 +565,8 @@ public static ModelNode parseAttributeValue(final String value, final boolean is
         final String trimmed = value == null ? null : value.trim();
         ModelNode node;
         if (trimmed != null) {
-            if (isExpressionAllowed && isExpression(trimmed)) {
-                node = new ModelNode(new ValueExpression(trimmed));
+            if (isExpressionAllowed && isExpression(value)) {
+                node = new ModelNode(new ValueExpression(value));
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/parsing/ParseUtils.java:569]: This should do the same test that the non-expression else block does, and use 'trimmed' unless attributeType is STRING or PROPERTY. An expression that's meant to resolve to a string convertible to some other type shouldn't have leading/trailing whitespace and we shouldn't stop being helpful about that.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: controller/src/test/java/org/jboss/as/controller/parsing/ParseUtilsTestCase.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ * @author Joerg Baesner (c) 2020 Red Hat Inc.
+ */
+public class ParseUtilsTestCase {
+
+    @Test
+    public void testExpressionsWithTrailingWhitespace() {
+
+        String propertyValue = "${key} ";
+        ModelType attributeType = ModelType.EXPRESSION;
```
Comment (bstansberry) [inline controller/src/test/java/org/jboss/as/controller/parsing/ParseUtilsTestCase.java:41]: @jbaesner I believe this should be ModelType.STRING

Comment (jbaesner) [reply controller/src/test/java/org/jboss/as/controller/parsing/ParseUtilsTestCase.java]: Yes, exactly, you're right!

File: controller/src/main/java/org/jboss/as/controller/parsing/ParseUtils.java (hunk 1 above)
Comment (jbaesner) [reply controller/src/main/java/org/jboss/as/controller/parsing/ParseUtils.java]: Ok, done...

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks, @jbaesner !

**Pull request merged**